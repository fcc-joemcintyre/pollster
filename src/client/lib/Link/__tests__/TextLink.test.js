/* eslint-env jest */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { TextLink, PlainTextLink } from '..';

const theme = {
};

test ('TextLink, to only, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <TextLink to='#'>text</TextLink>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('TextLink, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <TextLink to='#' subject='Hi'>text</TextLink>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainTextLink, to only, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainTextLink to='#'>text</PlainTextLink>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainTextLink, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainTextLink to='#' subject='Hi'>text</PlainTextLink>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainTextLink with color, to only, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainTextLink to='#' c='#0000ff'>text</PlainTextLink>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('PlainTextLink with color, default render', () => {
  const component = renderer.create (
    <ThemeProvider theme={theme}>
      <PlainTextLink to='#' subject='Hi' c='#0000ff'>text</PlainTextLink>
    </ThemeProvider>
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});
