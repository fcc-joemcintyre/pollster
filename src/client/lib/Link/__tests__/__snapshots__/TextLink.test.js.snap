// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainTextLink with color, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0000ff;
}

.c0:visited: {
  color: #0000ff;
}

<a
  c="#0000ff"
  className="c0"
  href="#"
  rel="noopener noreferrer"
  subject="Hi"
  target="_blank"
>
  text
</a>
`;

exports[`PlainTextLink with color, to only, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #0000ff;
}

.c0:visited: {
  color: #0000ff;
}

<a
  c="#0000ff"
  className="c0"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  text
</a>
`;

exports[`PlainTextLink, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
  href="#"
  rel="noopener noreferrer"
  subject="Hi"
  target="_blank"
>
  text
</a>
`;

exports[`PlainTextLink, to only, default render 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  className="c0"
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  text
</a>
`;

exports[`TextLink, default render 1`] = `
<a
  href="#"
  rel="noopener noreferrer"
  subject="Hi"
  target="_blank"
>
  text
</a>
`;

exports[`TextLink, to only, default render 1`] = `
<a
  href="#"
  rel="noopener noreferrer"
  target="_blank"
>
  text
</a>
`;
